File: 백준/Gold/26732. Agar．io/Agar．io.py
import sys
input = lambda: sys.stdin.readline().rstrip()

def solve(N, T):
	# 본인 세포의 크기
	cell = 2

	# (질량이) 가장 큰 세포가 되는 데 필요한 시간
	time = 0

	i = 0
	T.sort()

	# 먹을 수 있는 세포들
	edible = []

	while i < N:
		while i < N and cell > T[i]:
			edible.append(T[i])
			i += 1

		# 세포가 가장 큰 세포가 된 경우 (같은 경우 포함)
		if cell >= T[-1]:
			return time

		# 위의 경우를 제외하면, 더이상 먹을 수 없을 때 가장 큰 세포에 도달할 수 없다.
		if not edible:
			return "NIE"

		cell += edible.pop()
		time += 1

	# 모든 세포를 먹을 수 있으면 가장 큰 세포이다.
	return time

if __name__ == "__main__":
	N = int(input())
	T = [*map(int, input().split())]

	print(solve(N, T))

File: 백준/Gold/26732. Agar．io/Agar．io.cc
#pragma GCC optimize("Ofast")
#pragma GCC optimize("unroll-loops")

#include <bits/stdc++.h>
#define FAST_IO ios_base::sync_with_stdio(false); cin.tie(nullptr); cout.tie(nullptr)

using namespace std;
using ll = long long;

vector<ll> cells;
stack<ll> edible;

int main() {
    FAST_IO;

    // 1. Input
    int N; cin >> N;

    for (int i = 0; i < N; i++) {
        ll T; cin >> T; cells.push_back(T);
    }

    // 2. Solution
    int i = 0;
    ll cell = 2, time = 0;

    std::sort(cells.begin(), cells.end());

    while (i < N) {
        while (i < N && cell > cells[i]) edible.push(cells[i++]);

        if (cell >= cells.back()) {
            cout << time << "\n";
            return 0;
        }

        if (edible.empty()) {
            cout << "NIE" << "\n";
            return 0;
        }

        cell += edible.top(); edible.pop();
        time++;
    }

    cout << time << "\n";
    return 0;
}
